### PR TITLE
Implemented the Base Key Types

### PR DESCRIPTION
**Pull request by nothingalike**
PrivateKey, PublicKey, and Mnemonic types. Removed the BIP32 functions from the KeyService and converted to extension methods of the key types.

Issue #5

### CODE REVIEW

File: CardanoSharp.Wallet.Test/AddressTests.cs [hunk 1]
```diff
@@ -125,11 +121,10 @@ public void VerifyRewardAddress(string mnemonic, string stakingAddr)
         /// <param name="path"></param>
         /// <param name="rootKey"></param>
         /// <returns></returns>
-        private (byte[], byte[]) getKeyPairFromPath(string path, (byte[], byte[]) rootKey)
+        private (PrivateKey, PublicKey) getKeyPairFromPath(string path, PrivateKey rootKey)
```
Comment (tweakch) [inline CardanoSharp.Wallet.Test/AddressTests.cs:124]: Return KeyPair here (ie your suggested type)

**Review by tweakch — APPROVED** [review]
Looking good. I'd appreciate an effort to create a collection of test vectors for these classes.

File: CardanoSharp.Wallet.Test/AddressTests.cs (hunk 1 above)
Comment (nothingalike) [reply CardanoSharp.Wallet.Test/AddressTests.cs:124]: yeah I wrote this and thought about the Key (better name: KeyPair) type. Alright I get this added.

**Comment by nothingalike** [comment]
> Looking good. I'd appreciate an effort to create a collection of test vectors for these classes.

Awesome, yeah I'll add tests next.

**Comment by nothingalike** [comment]
Alright... I think I hit a bit of a wall.

1) What are some good test vectors for the new `PrivateKey` and `PublicKey`? Ultimately all the logic isn't necessarily new. Its just moved or restructured. We have existing tests around the Key logic: Generating Mnemonics and  Creating RootKeys. Once the `WalletPath` type is merged, I can update those tests to use the new types as well. @tweakch let me know your thoughts and any advice here

2) I tried to incorporate `SimpleExec` but the nature of how we pass parameters to the `cardano-address` binary presented some serious issues. Since it uses a pipe input, the only solutions I came across was to execute `cmd.exe` and execute your command as an input. I'm ok with this, but `cmd.exe` isn't cross platform from my understanding. @safestak-keith got any ideas for this one?

ultimately, I want to watch on #17 before merging this PR since the types added here would be impacted by `WalletPath`

**Pull request closed without merging**